go/registry/api/sanity_check.go

```diff
@@ -43,6 +54,28 @@ func (g *Genesis) SanityCheck(baseEpoch epochtime.EpochTime) error {
 	return SanityCheckNodes(logger, &g.Parameters, g.Nodes, seenEntities, runtimesLookup, true, baseEpoch)
 }
 
+// SanityCheckEntitiesStake examines if all seen entities have an associated
+// account and enough stake to be registered.
+func SanityCheckEntitiesStake(
+	logger *logging.Logger,
+	seenEntities map[signature.PublicKey]*entity.Entity,
+	stakeLedger map[signature.PublicKey]*staking.Account,
+	stakeThresholds map[staking.ThresholdKind]quantity.Quantity,
+) error {
+	for _, entity := range seenEntities {
+		acct := stakeLedger[entity.ID]
+		if acct == nil {
+			return fmt.Errorf("registry: sanity check failed: no account associated with entity with ID: %v", entity.ID)
+		}
+		entityThreshold := stakeThresholds[staking.KindEntity]
```

Comment:
could move this out of the loop. maybe even out of the function

Reply:
Good point, agreed.

Reply:
Done.